tools/bootstrap-generic-server.py

```diff
@@ -0,0 +1,216 @@
+
+            # check if user passed us an existing python file, if not: create a simple template
+            if not os.path.exists(module + ".py"):
+                with open(module + ".py", 'w+') as f:
+                    print(f"""
+import PyApplicationCore as ac
+
+class {module}(ac.ApplicationModule):
+    def __init__(self, owner):
+        super().__init__(owner, "{module}", "Please add description here")
+        
+    def mainLoop(self):
+        # Implementation goes here
+        pass
+        
+ac.app.{module} = {module}(ac.app)
+                          """, file=f)
+
+    def write(self):
+        with open('generic_device_server-config.xml', 'w') as f:
+            _pretty_write_xml(self.root, f)
+
+
+class DOOCS:
+    def __init__(self, catalogue, args):
+        self.catalogue = catalogue
+        self.facility, self.device, self.location = args.doocs_address.split("/")
+        self.hostname = args.hostname
+        self.rpc_number = args.doocs_rpc_number
+        self.mapped_device = args.device_name
+
+    def write_config(self):
+        with open('doocs-generic-chimeratk-server02.conf', 'w+') as f:
+            print(f"""
+eq_conf:
+ 
+oper_uid:       -1
+oper_gid:       1000
+xpert_uid:      0
+xpert_gid:      0
+ring_buffer:    10000
+memory_buffer:  500
+ 
+eq_fct_name:    "{self.hostname}._SVR"
+eq_fct_type:    1
+{{
+SVR.RPC_NUMBER:         {self.rpc_number}
+SVR.NAME:       "{self.hostname}._SVR"
+SVR.BPN:        6000
+SVR.FACILITY: "{self.facility}"
+SVR.DEVICE: "{self.device}"
+}}
+eq_fct_name:    "{self.location}"
+eq_fct_type:    10
+{{
+NAME:   "{self.location}"
+}}
+eq_fct_name:    "Debug"
+eq_fct_type:    10
+{{
+NAME:   "Debug"
+}}
+""", file=f)
+
+    def write_mapping(self):
+        root = ET.Element("{https://github.com/ChimeraTK/ControlSystemAdapter-DoocsAdapter}device_server")
+        location = ET.SubElement(root, "location", attrib={"name": self.location})
+        ET.SubElement(location, "import").text = "/" + self.mapped_device
+        location = ET.SubElement(root, "location", attrib={"name": "Debug"})
+        ET.SubElement(location, "import").text = "/"
+        with open("generic_chimeratk_server-DoocsVariableConfig.xml", "w+") as f:
+            _pretty_write_xml(root, f)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument("mapfile")
+    parser.add_argument('--device-name', default='DEV0')
+    parser.add_argument('--mapped-device-file', default='DEV0.xlmap', type=argparse.FileType('w', encoding='utf-8'))
+    parser.add_argument('--adapter', default='DOOCS', choices=['TANGO', 'DOOCS', 'OPC-UA'])
+    parser.add_argument('--add-python-module', default=[], nargs='+', dest="python_modules")
+    parser.add_argument('--hostname', default=socket.gethostname().upper())
+    parser.add_argument('--doocs-address', default='TEST.DOOCS/DEV0/DEV0.0')
+    parser.add_argument('--doocs-rpc-number', default='610498009')
+    args = parser.parse_args()
```

Comment:
Maybe - in the long run - a TUI based on Textual would be more helpful than a growing list of command line options?